packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -406,10 +406,31 @@ export class DocumentDeltaConnection
                 let isWebSocketTransportError = false;
                 try {
                     const description = error?.description;
-                    if (description && typeof description === "object") {
+                    const context = error?.context;
+
+                    if (context && typeof context === "object") {
+                        const statusText = context.statusText.code;
+
+                        // Instead of string-matching the ErrorMessage, Filter the Error Based on the ErrorCode: "DEPTH_ZERO_SELF_SIGNED_CERT"
+                        // Self-Signed Certificate ErrorCode Found in error.context
+                        if (statusText === "DEPTH_ZERO_SELF_SIGNED_CERT") {
+                            failAndCloseSocket(this.createErrorObject("connect_error", error, false));
+                            return;
+                        }
+                    }
+                    else if (description && typeof description === "object") {
+                        const errorCode = description.error.code;
```

Comment:
```suggestion
                        const errorCode = description.error?.code;
```
Let's protect ourselves against `description.error` being `undefined`.